### PR TITLE
functions for parent-Features inheritance and possible mutation

### PR DESCRIPTION
**Pull request by serlecu**
Used to feed the constructor of a new creature based on 2 parents.

### CODE REVIEW

**Review by CDKML — CHANGES_REQUESTED** [review]
do not commit settings json

File: Assets/Scripts/Genetics/rng_functions.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    {
+      direction = Random.Range(0, 2)-1;
+      ammount = Random.Range(0, 2);
+      return targetFeature + (direction * ammount % featureRange);
+    }
+
+    /** Inherit the same feature value as one of the parents or mutate */
+    int inheritFeature(int featureA, int featureB, int featureRange, int mutationProvability = 0)
+    {
+      provA = (100 - mutationProvability)/2;
```
Comment (CDKML) [inline Assets/Scripts/Genetics/rng_functions.cs]: change provA and provB names and probability (with b)

File: Assets/Scripts/Genetics/rng_functions.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class rng_functions : MonoBehaviour
```
Comment (CDKML) [inline Assets/Scripts/Genetics/rng_functions.cs]: Upper case rng_functions

File: Assets/Scripts/Genetics/rng_functions.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+{ 
+    /** We asume that features are designed as a range of integers */
+
+    /** Mutate means getting a different feature value than the parents */
+    int mutateFeature(int targetFeature, int featureRange)
```
Comment (CDKML) [inline Assets/Scripts/Genetics/rng_functions.cs]: uppercase

File: Assets/Scripts/Genetics/rng_functions.cs (hunk 1 above)
Comment (serlecu) [reply Assets/Scripts/Genetics/rng_functions.cs]: indeed

**Review by CDKML — APPROVED** [review]

**Pull request merged**